### ISSUE TITLE
QOL Improvement: Randomize shop item colors when placing a new shop.

### ISSUE THREAD

**Issue by grahambilley** [OpenRCT2/OpenRCT2#11113]
**OS:** [Windows 10]
**Version:** [0.2.4]
**Commit/Build:** [d645338]

This is an enhancement request. I'm putting this here per request of Aaron van Geffen.

Randomize the color of shop items (hats, balloons, shirts, umbrellas) when placing a new shop, instead of always defaulting to red (toggle on/off in settings).

Thank you,

**Comment by Umdlye** [comment]
This has been suggested before, you can check out the discussions there: #1438/#2154

**Comment by grahambilley** [comment]
I see these have been mentioned before. Referencing the other posts here. #1438/#2154

**Comment by Umdlye** [comment]
I deleted my earlier comment because I misread your suggestion, those two aren't the same as yours. They want shops to be able to sell a randomly colored item every sale. You just want the preset color to be different every time a shop is placed, right?

**Comment by grahambilley** [comment]
Thanks for clarifying. I suppose both suggestions solve the same problem. But I think having the default color be random is a better solution, since you still have the ability to color code areas of your park if you want to.

**Comment by ocalhoun6** [comment]
Seems like this would be _perfect_ for the kind of a thing a plugin should do.

**Comment by pizza2004** [comment]
Yeah, I would like the random option to be built into the game, basically being a new color to select from the color menu, but this would work very well as a plugin that just changes the color randomly when you place the shop.

**Comment by grahambilley** [comment]
I just want to clarify one thing on my intent for this enhancement (so this is just my opinion of course).

What I'm trying to avoid is the extra time and clicks changing the color of an item when I place a new shop. If there were a "random" color square that I could click, I would still have to pick that square every time I place a new shop, so it doesn't solve my problem. At that point, I may as well just pick a color that I like.

**Comment by gamerlenny** [comment]
Hey @grahambilley, I know this thread is kind of old, but I've been wanting the same feature you described, for the very same reasons, for a long time (which is actually how I stumbled into this topic the other day). So this weekend I decided to code a plugin that does exactly that: randomize the preset color when placing a new shop (instead of all balloons being blue by default, all umbrellas red, etc). 
Check it out : [Shop Item Color Manager](http://openrct2plugins.org/plugin/MDEwOlJlcG9zaXRvcnkzMzQ3MTk5NTc=/openrct2-shop-item-color-manager)

By default it gives a new, random color to each item when placing a new Hat / T-Shirt / Balloons / Souvenir stall or Info Kiosk, but it also allows you to choose a different preset color for each type of item, and have every new stall that sells that item set to that specific color instead of randomizing it every time. You can also apply the settings of your choice to all existing shops in a park and change them as often as you like in a few clicks...

It requires a recent develop version of OpenRCT2 though, since it uses the colorpicker element, which was recently added to the plugin API.

Cheers!

**Comment by grahambilley** [comment]
That is awesome! Thank you!
